serial-commands.c

```diff
@@ -60,9 +70,18 @@ int timed_read(int fd, char *buf, ssize_t buf_size) {
   if (select_result > 0) {
     printf("Time remaining: %i microseconds\n", timeout.tv_usec);
     printf("Is the FD ready? %i\n", FD_ISSET(fd, &input));
+
     memset(buf, 0, buf_size);
-    read_count = read(fd, buf, buf_size-1);
-    printf("Read %zi bytes of response: %s\n", read_count, buf);
+    do 
+    {
+      read_count = read (fd, buf+offset, buf_size);
```

Comment:
Formatting is a little weird here - space between `read` and its open paren.

I recently learned about [`clang-format`](http://clang.llvm.org/docs/ClangFormat.html)! I wonder what it would do on my terrible C code.